src/util/TimeUtil.js

```diff
@@ -93,4 +93,22 @@ export default {
       unit: convertMomentUnits(highestUnit),
     };
   },
+
+  /**
+   * @method formatDateToDeviceAssuranceGracePeriodExpiryLocaleString
+   * Conversion from a Date object to a locale string that mimics Okta's `short-with-timezone` format
+   * e.g. new Date(2024-09-05T00:00:00.000Z) -> 09/05/2024, 8:00 PM EDT
+   *
+   * @param {Date} date The Date object for the grace period expiry
+   * @param {LanguageCode} languageCode The user's language code / locale
+   * @return {String} The formatted `short-with-timezone` local string
+   */
+  formatDateToDeviceAssuranceGracePeriodExpiryLocaleString: (date, languageCode) => date.toLocaleString(languageCode, {
```

Comment:
Do we need to do any input validation or checks for null/undefined?

Reply:
Will move the check for an invalid `Date` object in here 👍 I just checked and `toLocaleString()` handles `null`/`undefined` values for the locale param so we shouldn't need to validate `languageCode`

Reply:
Actually looks like `toLocaleString()` only handles `undefined`. Will add some validation for other cases